pydantic/_internal/_generate_schema.py

```diff
@@ -644,14 +653,21 @@ def _get_first_two_args_or_any(self, obj: Any) -> tuple[Any, Any]:
         return args[0], args[1]
 
     def _post_process_generated_schema(self, schema: core_schema.CoreSchema) -> core_schema.CoreSchema:
-        schema.setdefault('metadata', {})[
-            NEEDS_APPLY_DISCRIMINATED_UNION_METADATA_KEY
-        ] = self._needs_apply_discriminated_union
+        metadata = schema.setdefault('metadata', {})
+        metadata[NEEDS_APPLY_DISCRIMINATED_UNION_METADATA_KEY] = self._needs_apply_discriminated_union
+        metadata['invalid'] = self._has_invalid_schema
```

Comment:
Should `'invalid'` maybe also become a constant?

Reply:
Done